frontend/app/routes/$lang+/_public+/apply+/$id+/applicant-information.tsx

```diff
@@ -67,7 +67,12 @@ export async function action({ context: { session }, params, request }: ActionFu
   // state validation schema
   const applicantInformationSchema: z.ZodType<ApplicantInformationState> = z
     .object({
-      socialInsuranceNumber: z.string().trim().min(1, t('apply:applicant-information.error-message.sin-required')).refine(isValidSin, t('apply:applicant-information.error-message.sin-valid')),
+      socialInsuranceNumber: z
+        .string()
+        .trim()
+        .min(1, t('apply:applicant-information.error-message.sin-required'))
+        .refine(isValidSin, t('apply:applicant-information.error-message.sin-valid'))
+        .refine((sin) => isValidSin(sin) && formatSin(sin, '') !== state.partnerInformation?.socialInsuranceNumber, t('apply:partner-information.error-message.sin-unique')),
```

Comment:
No need to call `isValidSin` before `formatSin` since it has been validate prior to this refine. I will format the partnerInformation SIN to compare, don't assume that partnerInformation SINI is using the same SIN format.

```suggestion
        .refine((sin) => state.partnerInformation?.socialInsuranceNumber !== undefined && formatSin(sin) !== formatSin(state.partnerInformation.socialInsuranceNumber), t('apply:partner-information.error-message.sin-unique')),
```

Reply:
All fair points!